GeneratorInterface/GenFilters/plugins/LHEGenericMassFilter.cc

```diff
@@ -0,0 +1,107 @@
+// system include files
+#include <memory>
+#include <iostream>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDFilter.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "SimDataFormats/GeneratorProducts/interface/LHEEventProduct.h"
+
+using namespace edm;
+using namespace std;
+
+//
+// class declaration
+//
+
+class LHEGenericMassFilter : public edm::EDFilter {
+public:
+  explicit LHEGenericMassFilter(const edm::ParameterSet&);
+  ~LHEGenericMassFilter() override;
+
+private:
+  bool filter(edm::Event&, const edm::EventSetup&) override;
+  void endJob() override;
+
+  // ----------member data ---------------------------
+
+  edm::EDGetTokenT<LHEEventProduct> src_;
```

Comment:
All class members can be made `const`, and the filter become a `global` one

Reply:
Actually, the counters prevented this change because they get incremented in the `filter` method. But since they are not necessary to have, I have removed them.